### ISSUE TITLE
Debian10: Official binary works while failed to run the binary built from source

### ISSUE THREAD

**Issue by Whishay** [Slicer/Slicer#4892]
I am running a Debain 10(buster) and the binary package for linux downloaded from official website works well.

Then I tried to build it from the source code and eventually I made it after installing various packages, but I can't not launch the 'Slice' app as expected as the one in the official binary package. I tried different options and here is the messages I got and I googled about the x11/wayland options but no one works. Can anyone help on this?

#./Slicer 
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
error: [/sdc/src/Slicer-local/Slicer-SuperBuild/Slicer-build/bin/./SlicerApp-real] exit abnormally - Report the problem.

#QT_QPA_PLATFORM=wayland  ./Slicer
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Using Wayland-EGL
error: [/sdc/src/Slicer-local/Slicer-SuperBuild/Slicer-build/bin/./SlicerApp-real] exit abnormally - Report the problem.

#QT_QPA_PLATFORM=wayland-egl  ./Slicer
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
Using Wayland-EGL
error: [/sdc/src/Slicer-local/Slicer-SuperBuild/Slicer-build/bin/./SlicerApp-real] exit abnormally - Report the problem

#XDG_SESSION_TYPE=x11  ./Slicer
error: [/sdc/src/Slicer-local/Slicer-SuperBuild/Slicer-build/bin/./SlicerApp-real] exit abnormally - Report the problem

## Environment
- Slicer version: Master(latest commit listed below)
- Operating system: Debian 10 with Gnome.
$ git log
commit da65fbce0d9c85f37418a3081bcd1429ba59bdec (HEAD -> master, origin/master, origin/HEAD)
Author: Jean-Christophe Fillion-Robin <jchris.fillionr@kitware.com>
Date:   Wed Mar 18 23:46:47 2020 -0400

    COMP: Fix configuration of extension with dependencies
    
    This commit ensures that setting EXTENSION_DEPENDS as either a list
    or a space separated string works as expected.
    
    Also to avoid increasing the complexity of SuperBuild extension template
    by systematically pre-processing the EXTENSION_DEPENDS variable, this
    commit also updates the template to suggest dependencies may only be
    specified as list.
    
    To summarize, the implementation supports specifying EXTENSION_DEPENDS
    in the following ways:
    - string: "ExtA ExtB ExtC" or "ExtA;ExtB;ExtC"
    - list: ExtA ExtB ExtC
    but the comment in the template recommend to only use list.
    
    Note that examples of extensions found in "Extensions/Testing" sub-directory
    have been re-generated building the "SlicerGenerateExtensionTemplates" target.

**Comment by Whishay** [comment]
Anyone can help have a look?  What kind of 'XDG_SESSION_TYPE' or 'QT_QPA_PALTFORM' should I choose?

#./Slicer
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
error: [/sdc/src/Slicer-local/Slicer-SuperBuild/Slicer-build/bin/./SlicerApp-real] exit abnormally - Report the problem.

**Comment by lassoan** [comment]
It does not seem like a Slicer issue, but more like a Qt configuration problem. Based on a quick web search, this is extensively discussed at many places on the web, which should help you to solve this. Let us know if you find a solution.

**Comment by RafaelPalomar** [comment]
Hello. @Whishay, what you are getting are warnings that might (or might not) have to do with the cause of the crash. Using a debugger, could possibly give better information on what is going on.

I've personally got nasty starting up crashes derived from the discrepancies between the OpenSSL library installed in the system and the OpenSSL library pulled from Slicer during build. Many linux systems are already using version 1.1, while Slicer was using 1.0 (I ignore if this has changed recently, though). 

To discard this end, and if I remember correctly, you can:
(1) build Slicer as normal
(2) once built, get into the python build directory and use cmake/ccmake to disable building python with SSL
(3) invoke make in the python build directory and later in the slicer build directory.
I can't recall whether any component of slicer needs to be disabled for this configuration to work (give it a try).

@lassoan, what do you think, does it make sense to try?

**Comment by jcfr** [comment]
> Many linux systems are already using version 1.1, while Slicer was using 1.0

Good point. We are overdue for updating the OpenSSL version used in Slicer. Now that we are also updating Qt5, this makes sense.

**Comment by jcfr** [comment]
Closing  this issue. 

The OpenSSL issue described above is expected to be fixed by https://github.com/Slicer/Slicer/pull/4992 integrating commit https://github.com/Slicer/Slicer/pull/4992/commits/3d8bc4ae374dcee2274375d23dc28445537d7143
